service/controller/resource/monitoring/prometheus/resource_test.go

```diff
@@ -36,69 +32,78 @@ func TestPrometheus(t *testing.T) {
 		}
 	}
 
-	c := unittest.Config{
-		OutputDir: outputDir,
-		T:         t,
-		TestFunc: func(v interface{}) (interface{}, error) {
-			testCluster, err := key.ToCluster(v)
-			if err != nil {
-				t.Fatal(err)
-			}
-			var secret runtime.Object
-			{
-				secret = &v1.Secret{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "cluster-certificates",
-						Namespace: key.Namespace(testCluster),
-					},
-					Data: map[string][]byte{
-						"token": []byte("my-token"),
-					},
-				}
-			}
+	for _, flavor := range unittest.ProviderFlavors {
+		outputDir, err := filepath.Abs("./test/" + flavor)
+		if err != nil {
+			t.Fatal(err)
+		}
 
-			var k8sClient k8sclient.Interface
-			{
-				c := k8sclient.ClientsConfig{
-					Logger:        logger,
-					SchemeBuilder: k8sclient.SchemeBuilder(v1.SchemeBuilder),
-				}
-				k8sClient, err = fake.NewClients(c, secret)
+		c := unittest.Config{
+			Flavor:    flavor,
+			OutputDir: outputDir,
+			T:         t,
+			TestFunc: func(v interface{}) (interface{}, error) {
+				testCluster, err := key.ToCluster(v)
 				if err != nil {
 					t.Fatal(err)
 				}
-			}
 
-			config := Config{
-				Address:            "http://prometheus/cluster",
-				Customer:           "Giant Swarm",
-				EvaluationInterval: "60s",
-				Installation:       "test-installation",
-				Pipeline:           "testing",
-				K8sClient:          k8sClient,
-				Provider: cluster.Provider{
-					Kind:   "aws",
-					Flavor: "vintage",
-				},
-				Region:          "onprem",
-				ImageRepository: "giantswarm/prometheus",
-				LogLevel:        "debug",
-				Registry:        "quay.io",
-				ScrapeInterval:  "60s",
-				Version:         "v2.28.1",
-			}
+				var secret runtime.Object
+				{
+					secret = &v1.Secret{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "cluster-certificates",
+							Namespace: key.Namespace(testCluster),
+						},
+						Data: map[string][]byte{
+							"token": []byte("my-token"),
+						},
+					}
+				}
 
-			return toPrometheus(context.Background(), v, config)
-		},
-		Update: *update,
-	}
-	runner, err := unittest.NewRunner(c)
-	if err != nil {
-		t.Fatal(err)
-	}
+				var k8sClient k8sclient.Interface
+				{
+					c := k8sclient.ClientsConfig{
+						Logger:        logger,
+						SchemeBuilder: k8sclient.SchemeBuilder(v1.SchemeBuilder),
+					}
+					k8sClient, err = fake.NewClients(c, secret)
+					if err != nil {
+						t.Fatal(err)
+					}
+				}
+
+				config := Config{
+					Address:            "http://prometheus/cluster",
+					Customer:           "Giant Swarm",
+					EvaluationInterval: "60s",
+					Installation:       "test-installation",
+					Pipeline:           "testing",
+					K8sClient:          k8sClient,
+					Provider: cluster.Provider{
+						Kind:   "aws",
```

Comment:
Provider is now correct for CAPI and vintage stays the same

Reply:
So it should not but with our change,we do not even use the kind anymore, this a remnant of vintage

Reply:
I could make this capa if flavor is capi, I just did not feel the need